pkg/controller/controller.go

```diff
@@ -243,18 +244,20 @@ func (c *Controller) monitor(watchVersion string) (<-chan *Event, <-chan error)
 				if err != nil {
 					if err == io.EOF { // apiserver will close stream periodically
 						c.logger.Debug("Apiserver closed stream")
+						// go to the next round
 						break
 					}
 
 					c.logger.Errorf("Received invalid event from API server: %v", err)
-					errCh <- err
-					return
+					// keep polling next event
+					continue
 				}
 
 				if st != nil {
 					if st.Code == http.StatusGone { // event history is outdated
-						errCh <- ErrVersionOutdated // go to recovery path
-						return
+						err = ErrVersionOutdated // go to recovery path
```

Comment:
Yes. Now I think keeping the `err` value is not really meaningful. Keep the controller running and logging unexpected errors are enough.